src/opam_cmd.ml

```diff
@@ -135,30 +135,87 @@ let classify_package ~package ~dev_repo ~archive ~pins () =
             | false -> (`Unknown host, tag) )
           | None -> err "dev-repo without host" ) )
 
-let check_if_dune ~repo package =
-  Exec.get_opam_depends ~repo (string_of_package package)
-  >>| List.exists (fun l -> l = "jbuilder" || l = "dune")
 
-let get_opam_info ~repo ~pins package =
-  Exec.get_opam_dev_repo ~repo (string_of_package package)
-  >>= fun dev_repo ->
-  Exec.get_opam_archive_url ~repo (string_of_package package)
-  >>= fun archive ->
-  let dev_repo, tag = classify_package ~package ~dev_repo ~archive ~pins () in
-  check_if_dune ~repo package
-  >>= fun is_dune ->
-  Logs.info (fun l ->
+let parse_string ~field ~package v =
+  let open OpamParserTypes in
+  match v with
+  | Some (String (_, v)) -> Ok v
+  | Some (List (_, [])) -> Ok ""
+  | None -> Ok ""
+  | _ ->
+      R.error_msg
+        (Fmt.strf
+           "Unable to parse opam string.\n\
+            Try `opam show --normalise -f %s: %s`"
+           field package)
+
+let parse_dev_repo = parse_string ~field:"dev-repo"
+
+let parse_archive_url ~package v =
+  parse_string ~field:"url.src" ~package v
+  >>= function
+  | "" -> Ok None
+  | uri -> Ok (Some uri)
+
+let parse_opam_depends ~package v =
+  let open OpamParserTypes in
+  match v with
+  | Some (List (_, vs)) ->
+      let ss =
+        List.fold_left
+          (fun acc -> function String (_, v) -> v :: acc
+            | Option (_, String (_, v), _) -> v :: acc | _ -> acc )
+          [] vs
+      in
+      Logs.debug (fun l ->
+          l "Depends for %s: %s" package (String.concat ~sep:" " ss) ) ;
+      Ok ss
+  | None -> Ok []
+  | _ ->
+      R.error_msg
+        (Fmt.strf
+           "Unable to parse opam depends for %s\n\
+            Try `opam show --normalise -f depends: %s` manually"
+           package package)
+
+
+let get_opam_info ~repo ~pins packages =
+  let fields = ["dev-repo"; "url.src"; "depends"] in
+  Exec.get_opam_fields ~repo fields packages
+   >>= fun data ->
+  let packages = List.map
+    (fun pkg ->
+      let name = pkg.name in
+      let archive_url_line = Types.StrMap.find_opt (name^".url.src") data in
```

Comment:
What would you think about extracting an abstract type for those `opam show` results that you could query with a more suited API, something in the lines of:
```ocaml
let archive_url_line = Opam_show_result.get ~package:pkg.name ~field:"url.src" data in
```
?

Reply:
Indeed that would be more readable